File: core/LinkedListCycle/LinkedListCycle.py
from typing import Optional

class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None


def hasCycle(head: Optional[ListNode]) -> bool:
    while head:
        if head.val > 100000:
            return True
        head.val += 2000001
        head = head.next

    return False

File: core/LinkedListCycle/LinkedListCycle.java
package com.dbc;

public class LinkedListCycle {
    class ListNode {
        int val;
        ListNode next;

        ListNode(int x) {
            val = x;
            next = null;
        }
    }

    public boolean hasCycle(ListNode head) {
        while (head != null) {
            if (head.val > 100000) return true;
            head.val += 200001;
            head = head.next;
        }
        return false;
    }
}
